src/test/objectstore/Allocator_test.cc

```diff
@@ -47,6 +47,23 @@ TEST_P(AllocTest, test_alloc_init)
   ASSERT_EQ(alloc->get_free(), (uint64_t) 0);
 }
 
+TEST_P(AllocTest, test_init_add_free)
+{
+  int64_t block_size = 1024;
+  int64_t capacity = 4 * 1024 * block_size;
+
+  {
+    init_alloc(capacity, block_size);
+
+    auto free = alloc->get_free();
+    alloc->init_add_free(block_size, 0);
+    ASSERT_EQ(free, alloc->get_free());
+
+    alloc->init_rm_free(block_size, 0);
```

Comment:
need to update `_remove_from_tree()` and `AvlAllocator::_try_remove_from_tree()` to return early if `size == 0`.

Reply:
@tchaikov - actually the approach is to handle zero length earlier at init_rm_free, just missed that stuff for HybridAllocator. Fixed mw, thanks!